crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -180,23 +184,31 @@ impl ModuleSubscriptionActor {
 
     fn remove_subscriber(&mut self, client_id: ClientActorId) {
         self.subscriptions.retain_mut(|sub| {
-            sub.remove_subscriber(client_id);
-            !sub.subscribers().is_empty()
+            let mut subscription = sub.write();
+            subscription.remove_subscriber(client_id);
+            !subscription.subscribers().is_empty()
         })
     }
 
-    async fn _broadcast_commit_event(&mut self, mut event: ModuleEvent, tx: &mut Tx) -> Result<(), DBError> {
-        let futures = FuturesUnordered::new();
-        let auth = AuthCtx::new(self.owner_identity, event.caller_identity);
-
-        for subscription in &mut self.subscriptions {
+    async fn broadcast_commit_event(&mut self, event: ModuleEvent) -> Result<(), DBError> {
```

Comment:
Resolving this comment since what I said is wrong. It is indeed run in an async manner, which means we already have a consistency issue, but we can solve that in a separate patch.